sdarq/backend/src/app.py

```diff
@@ -804,50 +799,45 @@ def get_sec_controls_service():
     user_email = request.headers.get('X-Goog-Authenticated-User-Email')
 
     if request.headers.get('Content-Type') != 'application/json':
-        return Response(json.dumps(
-            {'statusText': 'Bad Request'}), status=400, mimetype='application/json')
+        return Response(json.dumps({'statusText': 'Bad Request'}),
+                        status=400, mimetype='application/json')
 
-    if re.match(pattern, service_name, re.IGNORECASE):
-        try:
-            doc_ref = db.collection(security_controls_firestore_collection).document(
-                service_name.lower())
-            doc = doc_ref.get()
-            if doc.exists:
-                return doc.to_dict()
-            else:
-                message = """
-                This service does not exist!
-                """
-                logging.info(
-                    "User %s requested to read security controls of a service that does not exist.",
-                    user_email)
-                return Response(json.dumps(
-                    {'statusText': message}), status=404, mimetype='application/json')
-        except Exception as error:
-            error_message = f"Exception /get_sec_controls_service enspoint: {error}"
-            logging.warning(error_message)
-            message = """
-            There is something wrong with the input! Server did not respond correctly to your request!
-            """
-            status_code = 404
-            return Response(json.dumps(
-                {'statusText': message}), status=status_code, mimetype='application/json')
-    else:
-        message = """
-        Please enter a valid value for your service name! Contact AppSec team for more information.
-        """
-        logging.info(
-            "User %s did not provide a valid value for the service name to read security controls.",
-            user_email)
+    json_data = request.get_json()
+    service_name = json_data.get('service')
+
+    if not service_name or not re.match(r'^[a-zA-Z0-9][a-zA-Z0-9-_ ]{1,28}[a-zA-Z0-9]$', service_name):
+        message = 'Please enter a valid value for your service name! Contact team for more information.'
+        logging.info('User %s did not provide a valid value for the service name to read security controls.', user_email)
         return Response(json.dumps({'statusText': message}),
                         status=404, mimetype='application/json')
 
+    try:
+        service_name_lowercase = service_name.lower()
+        doc_ref = db.collection(security_controls_firestore_collection).document(service_name_lowercase)
+        doc = doc_ref.get()
+        if doc.exists:
+            return doc.to_dict()
+        else:
+            message = 'This service does not exist!'
+            logging.info('User %s requested to read security controls of a service that does not exist.', user_email)
+            return Response(json.dumps({'statusText': message}), status=404, mimetype='application/json')
```

Comment:
An editing UI and/or pick list (at some point in the future) would help avoid too much of this type of user error.

Reply:
Thanks, Tom! Will take a look and see how can I improve this part.